src/persist/src/s3.rs

```diff
@@ -92,6 +94,19 @@ impl S3BlobConfig {
             ))
         }
 
+        // TODO: our timeouts will all be redone when https://github.com/awslabs/smithy-rs/pull/1740
+        // is released. it contains a much more straightforward and robust way of thinking about
+        // AWS client timeouts.
+        loader = loader.timeout_config(
+            timeout::Config::new().with_api_timeouts(
+                timeout::Api::new()
+                    // maximum time allowed for a top-level S3 API call (including internal retries)
+                    .with_call_timeout(TriState::Set(Duration::from_secs(180)))
+                    // maximum time allowed for a single network call
+                    .with_call_attempt_timeout(TriState::Set(Duration::from_secs(90))),
```

Comment:
this is very conservatively set. since we use multipart for large blobs, our mean get/set time is 50ms. 90s is super long, but it should be better than none